### PR TITLE
Bump to rabbitmq v2.6.0 structs, also new bundle

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

**Review by bshephar — APPROVED** [review]
lgtm

**Comment by abays** [comment]
> @dprince: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> Test name 	Commit 	Details 	Required 	Rerun command
> ci/prow/openstack-operator-build-deploy-kuttl 	[506d9e0](https://github.com/openstack-k8s-operators/openstack-operator/commit/506d9e0d615ff6682bb73c1fcc24a89e7296dc1e) 	[link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openstack-k8s-operators_openstack-operator/679/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1759840575214850048) 	true 	`/test openstack-operator-build-deploy-kuttl`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openstack-k8s-operators&repo=openstack-operator&pr=679). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:dprince).

Looks like something isn't parsing correctly:

```
time="2024-02-20T07:37:20Z" level=fatal msg="initializing source docker://pull:latest: reading manifest latest in docker.io/library/pull: requested access to the resource is denied"
```

File: hack/pin-bundle-images.sh [hunk 1]
```diff
@@ -86,5 +86,6 @@ for MOD_PATH in $(go list -mod=readonly -m -json all | jq -r '. | select(.Path |
 done
 # append the rabbitmq URL only if we aren't in Dockerfile mode
 if [ -z "$DOCKERFILE" ]; then
-    echo -n ",quay.io/openstack-k8s-operators/rabbitmq-cluster-operator-bundle@sha256:2a09973faa91ae6d3913d61cc0ae7e124240c27f886563de87a2fbdd3aec6ad0"
+    # pin rabbit to sha256 for the e2d413c102bc73d4b5e186d1d1b1f9bf728701e1 tag (v2.6.0)
+    echo -n ",pull quay.io/openstack-k8s-operators/rabbitmq-cluster-operator-bundle@sha256:b3e37b529e76faf5b140021b3e5fd9ba5ef8cfa7c26e74845cf47d075e204906"
```
Comment (abays) [inline hack/pin-bundle-images.sh:90]: Should this `pull` be here?  It seems to be showing up in the CI failures as if it were the actual name of an image:

```
time="2024-02-19T18:43:12Z" level=fatal msg="initializing source docker://pull:latest: reading manifest latest in docker.io/library/pull: requested access to the resource is denied"
```

Comment (dprince) [reply hack/pin-bundle-images.sh:90]: ugh, copy paste error on my part from quay.io

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**